File: Data-Structures-and-Algorithms/maxavgsubarray/maxavgsubarray.js
var findMaxAverage = function(nums, k) {
    let cursum = 0;
    let maxsum = 0;
    for (var i = 0; i < k; i++) {
        cursum += nums[i];
    };
    maxsum = cursum;
    for (var j = k; j < nums.length; j++) {
        cursum += nums[j] - nums[j-k]
        maxsum = Math.max(maxsum, cursum);
    };
    
    return maxsum*1.0/k;
};

var arr = [-1];
var k  = 1

console.log(findMaxAverage(arr, k));
console.log(arr);

// O(N) for time (N/2 swaps)
// O(1) for space (no recursion)

File: Data-Structures-and-Algorithms/maxavgsubarray/maxavgsubarray.py
class Functions(object):
    def findMaxAverage(self, nums, k):
        maxsum = 0
        cursum = 0
        for i in range(k):
            cursum += nums[i]
        maxsum = cursum
        for j in range(k, len(nums)):
            cursum += nums[j]- nums[j-k]
            maxsum = max(cursum, maxsum)
        return maxsum*1.0/k
        

    
arr = [-1,2,-4]
k = 2
print(arr)
print(Functions().findMaxAverage(arr,k))


# O(N) for time (N steps to iterate through)
# O(1) for space (constant extra space)